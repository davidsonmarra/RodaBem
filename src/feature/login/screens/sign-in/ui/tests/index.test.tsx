import React from 'react';
import {create} from 'react-test-renderer';
import SignInContainer from '..';
import {Button, Input} from '@components';

const mockOnSubmit = jest.fn();
const mockErrors = undefined;
const mockClearErrors = jest.fn();
const mockOnSignUp = jest.fn();

const containerInstance = ({
  control,
  onSubmit = mockOnSubmit,
  errors = mockErrors,
  clearErrors = mockClearErrors,
  onSignUp = mockOnSignUp,
}: any) =>
  create(
    <SignInContainer
      control={control}
      onSubmit={onSubmit}
      errors={errors}
      clearErrors={clearErrors}
      onSignUp={onSignUp}
    />,
  );

describe('SignInContainer', () => {
  it('should render correctly', () => {
    const tree = containerInstance({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onSubmit', () => {
    const instance = containerInstance({}).root;
    const buttonsInstance = instance.findAllByType(Button);
    buttonsInstance[1].props.onPress();
    expect(mockOnSubmit).toHaveBeenCalled();
  });

  it('should call onSignUp', () => {
    const instance = containerInstance({}).root;
    const buttonsInstance = instance.findAllByType(Button);
    buttonsInstance[2].props.onPress();
    expect(mockOnSignUp).toHaveBeenCalled();
  });

  it('should call clearErrors', () => {
    const instance = containerInstance({}).root;
    const inputsInstance = instance.findAllByType(Input);
    inputsInstance[0].props.clearError();
    inputsInstance[1].props.clearError();
    expect(mockClearErrors).toHaveBeenCalledTimes(2);
  });

  it('should render error messages', () => {
    const errors = {
      email: {
        message: 'Email is required',
      },
      password: {
        message: 'Password is required',
      },
    };
    const instance = containerInstance({errors}).root;
    const inputsInstance = instance.findAllByType(Input);
    expect(inputsInstance[0].props.error).toBe('Email is required');
    expect(inputsInstance[1].props.error).toBe('Password is required');
  });

  it('should not render error messages', () => {
    const instance = containerInstance({}).root;
    const inputsInstance = instance.findAllByType(Input);
    expect(inputsInstance[0].props.error).toBeUndefined();
    expect(inputsInstance[1].props.error).toBeUndefined();
  });
});
