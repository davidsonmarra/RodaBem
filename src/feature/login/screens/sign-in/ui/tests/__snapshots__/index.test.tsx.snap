// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInContainer should render correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F1F1F1",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingHorizontal": 24,
      "paddingVertical": 16,
    }
  }
>
  <View>
    <Text
      style={
        [
          {
            "color": "#121212",
            "fontSize": 24,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Logo
    </Text>
  </View>
  <View
    style={
      {
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#222222",
            "fontSize": 16,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Entre na sua conta
    </Text>
    <View
      style={
        {
          "height": 24,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#222222",
              "fontSize": 14,
              "fontWeight": "normal",
            },
            {
              "width": "100%",
            },
          ]
        }
      >
        E-mail
      </Text>
      <View
        style={
          {
            "height": 8,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#595959",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 62,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Digite seu e-mail"
          style={
            {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "height": 16,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#222222",
              "fontSize": 14,
              "fontWeight": "normal",
            },
            {
              "width": "100%",
            },
          ]
        }
      >
        Senha
      </Text>
      <View
        style={
          {
            "height": 8,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#595959",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 62,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Digite sua senha de acesso"
          secureTextEntry={true}
          style={
            {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "height": 24,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": "auto",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#E30B0B",
              "fontSize": 18,
              "fontWeight": "bold",
              "padding": 8,
            },
            undefined,
          ]
        }
      >
        Esqueci a senha
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#E30B0B",
          "borderRadius": 12,
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 8,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
              "padding": 8,
            },
            undefined,
          ]
        }
      >
        Entrar
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderRadius": 12,
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 8,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#E30B0B",
              "fontSize": 18,
              "fontWeight": "bold",
              "padding": 8,
            },
            undefined,
          ]
        }
      >
        Cadastrar
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
